File: backend/schemas/material/batch_code.py
from pydantic import BaseModel, Field
from datetime import date


class BatchCodeGenerateRequest(BaseModel):
    """批次编码生成请求模型"""
    material_id: int = Field(..., description="器材ID")
    batch_date: date = Field(..., description="日期")


class BatchCodeGenerateResponse(BaseModel):
    """批次编码生成响应模型"""
    batch_code: str = Field(..., description="生成的批次编码")
    material_code: str = Field(..., description="器材编码")
    batch_date: date = Field(..., description="日期")
    sequence: int = Field(..., description="流水号")

File: frontend/src/services/types/batch_code.ts
// 批次编码生成相关类型定义

/** 批次编码生成请求参数 */
export interface BatchCodeGenerateRequest {
  material_id: number;
  batch_date: string; // 日期格式: YYYY-MM-DD
}

/** 批次编码生成响应数据 */
export interface BatchCodeGenerateResponse {
  batch_code: string; // 生成的批次编码
  material_code: string; // 器材编码
  batch_date: string; // 日期
  sequence: number; // 流水号
}

/** 批次编码生成API响应类型别名 */
export type BatchCodeGenerateResponseType = BatchCodeGenerateResponse;